one_fm/one_fm/doctype/attendance_check/attendance_check.py

```diff
@@ -301,10 +301,16 @@ def create_attendance_check(attendance_date=None):
         if not attendance_date:
             attendance_date = add_days(today(), -1)
 
+        employee_with_shift_permission = [i.employee for i in frappe.db.get_list('Shift Permission', {
+            'date': attendance_date
+        }
+        )]
+
         absentees = frappe.get_all("Attendance", filters={
             'docstatus':1,
             'status':'Absent',
-            'attendance_date':attendance_date},
+            'attendance_date':attendance_date,
+            'employee':["not in", employee_with_shift_permission]},
```

Comment:
so instead of declaring it as a variable you could just writhe the query here, unless you are using it somewhere else

Reply:
I used variable to make code clean and clearer for someone else. If I add everything in a single it may look ambiguous for new developer